src/app/project/project.component.spec.ts

```diff
@@ -59,12 +36,137 @@ describe('ProjectComponent', () => {
+    component = undefined;
   });
+  describe('when there is project and build data available', () => {
+    beforeEach(() => {
+      project = {
+        id: 'brigade-1234',
+        name: 'kashti/localdev',
+        repo: {
+          name: 'github.com/kashti/localdev',
+          cloneURL: 'https://github.com/kashti/localdev.git'
+        },
+        defaultScript: '',
+        kubernetes: {
+          namespace: 'default',
+          vcsSidecar: '',
+          buildStorageSize: '50Mi'
+        },
+        github: {
+          baseURL: '',
+          uploadURL: ''
+        },
+        secrets: {},
+        worker: {
+          registry: '',
+          name: '',
+          tag: '',
+          pullPolicy: ''
+        }
+      };
 
-  it('should create when no project data is available', () => {
-    project = {};
-    expect(component).toBeTruthy();
+      builds = [
+        {
+          id: '01c0mar8tk6daeabj7fhmcskc6',
+          project_id: 'brigade-1234',
+          type: 'push',
+          provider: 'github',
+          revision: {
+            commit: '',
+            ref: ''
+          },
+          payload: '',
+          script: '',
+          worker: {
+            id: 'brigade-worker-01c0mar8tk6daeabj7fhmcskc6-ad0703ac',
+            build_id: '01c0mar8tk6daeabj7fhmcskc6',
+            project_id: 'brigade-29d38c7477ecee18e184b69bec354fc350605c51bc16d4dd2b6073',
+            start_time: '2018-02-26T22:57:27Z',
+            end_time: '2018-02-26T22:57:35Z',
+            exit_code: 1,
+            status: 'Failed'
+          }
+        },
+        {
+          id: '01c0matkmjpjkrbfcnhfyjerfh',
+          project_id: 'brigade-abcd',
+          type: 'pull_request',
+          provider: 'github',
+          revision: {
+            commit: '',
+            ref: ''
+          },
+          payload: '',
+          script: '',
+          worker: {
+            id: 'brigade-worker-01c0matkmjpjkrbfcnhfyjerfh-ad0703ac',
+            build_id: '01c0matkmjpjkrbfcnhfyjerfh',
+            project_id: 'brigade-29d38c7477ecee18e184b69bec354fc350605c51bc16d4dd2b6073',
+            start_time: '2017-12-23T07:32:08Z',
+            end_time: '2017-12-23T07:32:21Z',
+            exit_code: 0,
+            status: 'Success'
+          }
+        }
+      ];
+    });
+
+    it('should create the component', () => {
+      expect(component).toBeTruthy();
+    });
+
+    it('should set the project', async(() => {
+      expect(component.project).toEqual(project);
+      expect(fixture.nativeElement.querySelector('h1')
+        .textContent).toContain('kashti/localdev');
+    }));
+
+    it('should set the builds', async(() => {
+      expect(component.builds).toEqual(builds);
+    }));
+  });
+
+  describe('when project and build data is unavailable', () => {
+    beforeEach(() => {
+      project = undefined;
+      builds = undefined;
+    });
+
+    it('should create the component', () => {
+      expect(component).toBeTruthy();
+    });
```

Comment:
Interestingly (to me) this expectation must be run before in any describe block that tests component properties.